src/main/java/org/samoxive/safetyjim/discord/DiscordShard.kt

```diff
@@ -347,6 +348,49 @@ class DiscordShard(private val bot: DiscordBot, shardId: Int, sessionController:
             }
         }
 
+        if (guildSettings.invitelinkremover){
+            val blacklistedHosts = arrayOf("discord.gg/")
+            for (blacklistedHost in blacklistedHosts) {
+                if (member.contains(blacklistedHost)) {
+
+                    val kickUser = member;
+
+                    val embed = EmbedBuilder()
+                    embed.setTitle("Kicked from " + guild.name)
+                    embed.setColor(Color(0x4286F4))
+                    embed.setDescription("You were kicked from " + guild.name)
+                    embed.addField("Reason: For username that contains an invite link" , false)
+                    embed.setFooter("Kicked by " + bot, null)
+                    embed.setTimestamp(now.toInstant())
+
+                    kickUser.sendDM(embed.build())
+
+                    try {
+                        val auditLogReason = String.format("Kicked by %s - %s", user.getUserTagAndId(), reason)
+                        controller.kick(kickUser.id, auditLogReason).complete()
+                        message.successReact(bot)
+
+                        val record = transaction {
+                            JimKick.new {
+                                userid = kickUser.id
+                                moderatoruserid = user.id
+                                guildid = guild.id
+                                kicktime = now.time / 1000
+                                this.reason = reason
+                            }
+                        }
+
+                        message.createModLogEntry(bot, shard, kickUser, "For username that contains an invite link", "kick", record.id.value, null, false)
+                        channel.trySendMessage("Kicked " + kickUser.getUserTagAndId())
```

Comment:
Where does `channel` object come from?

Reply:
Gradle has some problems on my computer that I cannot sync and test. It's all red, and I am using idea as notepad. So I forgot, let me re-edit